Services/Component/classes/class.ilPlugin.php

```diff
@@ -857,7 +857,13 @@ protected function beforeUninstall()
 				$ilDB->manipulate("DELETE FROM lng_modules".
 					" WHERE module = ".$ilDB->quote($prefix, "text"));
 			}
-			
+
+			// clear the event listening
+			include_once("Services/Component/classes/class.ilPluginReader.php");
+			$reader = new ilPluginReader($this->getDirectory() . '/plugin.xml',
+				$this->getComponentType(), $this->getComponentName(), $this->getSlotId(), $this->getPluginName());
+			$reader->clearEvents();
```

Comment:
Please move this to a function `clearEventListening`, that would spare the comment above and is good for general fitness.